pinot-core/src/main/java/org/apache/pinot/core/operator/query/NonScanBasedAggregationOperator.java

```diff
@@ -144,6 +148,27 @@ protected AggregationResultsBlock getNextBlock() {
           result = getDistinctCountSmartHLLResult(Objects.requireNonNull(dataSource.getDictionary()),
               (DistinctCountSmartHLLAggregationFunction) aggregationFunction);
           break;
+        case DISTINCTCOUNTULL:
+          result = getDistinctCountULLResult(Objects.requireNonNull(dataSource.getDictionary()),
+              ((DistinctCountULLAggregationFunction) aggregationFunction)
+                  .getP());
+          break;
+        case DISTINCTCOUNTSMARTULL:
+          result = getDistinctCountSmartULLResult(Objects.requireNonNull(dataSource.getDictionary()),
+              (DistinctCountSmartULLAggregationFunction) aggregationFunction);
+          break;
+        case DISTINCTCOUNTRAWULL: {
+          Dictionary dictionary = Objects.requireNonNull(dataSource.getDictionary());
+          int p =
+              ((DistinctCountULLAggregationFunction) aggregationFunction)
+                  .getP();
+          if (dictionary.getValueType() == FieldSpec.DataType.BYTES) {
```

Comment:
Why do we need to specialize BYTES type here? It should be the same as DISTINCTOUNSMARTHLL